plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -122,9 +123,19 @@ export class DeploymentManager {
     this.contractsCache = null;
   }
 
+  /* Conditionally executes an action */
+  async idempotent(
```

Comment:
Could be, didn't really think about that yet since didn't have use for the return on any of them

Reply:
My suggestion is more in re: making sure that each invoked function must match on that type, but if it's just for side-effects and not return type, then yeah, could just be Promise<void> then, right?

Reply:
Yeah I changed it locally already to have the return value be a T of whatever action is like you suggested, but yeah it could probably just be `Promise<void>` but maybe there'll be a reason to pass it through before this is done